### PR TITLE
chor: implements #12 - Usage in NS-NG2 app

### PR DESCRIPTION
**Pull request by m-abs**
implements #12

This adds an angular directive based on nativescript-angular-3.1.0's list-view-comp and the grid-view.module.ts, which an angular developer will have to import in his/hers own NgModule.

Notes:
- `grunt build` now use ngc, angular's extension to tsc. This generates the `metadata.json`-files needed for AOT/bundle builds.
- There is a new demo-project in the `demo-ng`-folder.
- I'm experience a minor? bug on android. I hope to find a solution tomorrow but I could use some input on it.

*Steps to reproduce:*
1. Run the demo-ng app on android
2. Tap on any item, you should now have navigated to a new page.
3. Click back.
4. All the items now have re-positioned, so each is on a single line.
5. Do a small drag gesture and they'll move back into view.

### CODE REVIEW

**Comment by PeterStaev** [comment]
@m-abs awesome work! Have some comments:
1. Lets move all angular related files to a subfolder `angular` similar to how it is in the Drop Down widget so we keep things isolated: 
https://github.com/PeterStaev/NativeScript-Drop-Down/tree/master/angular
2. Lets use `ngc` to compile only the angular files and not the rest, again as a references you see how it is made in the Drop Down widget:
https://github.com/PeterStaev/NativeScript-Drop-Down/blob/master/gruntfile.js#L50
https://github.com/PeterStaev/NativeScript-Drop-Down/blob/master/tsconfig.aot.json
3. Lets name the Module `GridViewModule` instead of `TNSGrdiViewModule`. I do not like to prefix public stuff as i want to keep things as similar to the core modules as possible. 
4. Lets remove the `ngx-` prefix from the component file. 

I will try to look up the problem you explained a bit later today, but I'm not a big angular expert so not sure if I will be able to help much 😞

**Comment by m-abs** [comment]
I made the corrections you suggested.

I still haven't fixed the issue I mentioned before.

**Comment by m-abs** [comment]
I found a workaround.

It is a bit difficult to explain, but I'll try.

I think, it is a timing issue cause by the `refresh()` here:
https://github.com/PeterStaev/NativeScript-Grid-View/blob/master/grid-view.android.ts#L134-L144

I tried logging the values.
* this._innerWidth === 0
* this._effectiveColWidth === undefined
* spanCount === 1

Side note: `this.isLayoutValid` === false.

In my case the values should be:
* this._innerWidth === 1080
* this._effectiveColWidth === 540
* spanCount === 2

When I get these values:
`this.isLayoutValid` === true

I found two ways to workaround the issue, without touching the non-angular code
1. Wait for `gridView.isLayoutValid` === true and call `refresh` again.
2. Add a `loadedEvent` on the `gridView` (but only on android) that calls refresh.

Option 2 was the cleanest of them so I decided on that.

**Comment by m-abs** [comment]
I may have found a better way to solve it.

If I move the line:
https://github.com/PeterStaev/NativeScript-Grid-View/blob/master/grid-view.android.ts#L276
To the bottom of the function of the `owner.onLayout()` is called before `super.onLayout()`, the problem seems to go away.

Would that be a solution, or would that break something else?

File: README.md [hunk 1]
```diff
@@ -33,6 +33,44 @@ You need to add `xmlns:gv="nativescript-grid-view"` to your page tag, and then s
...
+      <StackLayout [nsRouterLink]="['/item', item.id]" borderColor="blue" borderWidth="2" borderRadius="5" verticalAlignment="stretch" class="list-group-item">
+        <Label verticalAlignment="center" [text]="item.name" class="list-group-item-text"></Label>
+      </StackLayout>
+    </ng-template>
+  </GridView>
+```
+
+In your app-module:
+```typescript
+import { TNSGridViewModule } from 'nativescript-grid-view/grid-view.module';
```
Comment (PeterStaev) [inline README.md]: This and below should be changed to use the new name `GridViewModule`

File: demo-ng/app/app.module.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { ItemsComponent } from "./item/items.component";
+import { ItemDetailComponent } from "./item/item-detail.component";
+
+// Uncomment and add to NgModule imports if you need to use two-way binding
+// import { NativeScriptFormsModule } from "nativescript-angular/forms";
+
+// Uncomment and add to NgModule imports  if you need to use the HTTP wrapper
+// import { NativeScriptHttpModule } from "nativescript-angular/http";
+
+import { GridViewModule } from 'nativescript-grid-view/angular/index';
```
Comment (PeterStaev) [inline demo-ng/app/app.module.ts]: Won't this work if we import it only like this (i.e. w/o the `index`):
```ts
import { GridViewModule } from 'nativescript-grid-view/angular';
```

**Comment by PeterStaev** [comment]
Hey @m-abs , let's try to move the `super` call to the bottom of the method. I'm just used to first call the the base class method, that's why I've put it at the start. Since this fixes the problem you described I think we can safely try it out! 😄

File: README.md (hunk 1 above)
Comment (m-abs) [inline README.md]: In the confusion I forgot to update the README.

File: demo-ng/app/app.module.ts (hunk 2 above)
Comment (m-abs) [inline demo-ng/app/app.module.ts]: Yes, bad habit :)

**Pull request merged**